### PR TITLE
Use Ouranos cookiecutter-pypackage recipe with cruft to generate base package

### PR DESCRIPTION
**Pull request by Zeitsperre**
This PR adds a package based on an Ouranosinc fork of audrefeldroy/cookiecutter-pypage using a few nice `tox` and `travis-ci` configurations. This recipe offers a few more options for performing `tox`-specific test locally and implements code style checks using `black` with python3.6+ compatibility.

With `Cruft`, periodic updates of this boilerplate code can be performed as the Ouranosinc branch is updated/optimized. This PR was generated from the cookiecutter found at commit https://github.com/Ouranosinc/cookiecutter-pypackage/commit/2bb9ee8b4ac8c2beaff6f71aa16a6125982e07fb. This is all tracked by `Cruft`, therefore nothing needs to be done, I'm just leaving this here as a reference.

Initialized values are based on what I assumed were desired, but if anything needs changing, I can regenerate with different values (they are tracked in `.cruft.json`). Any generic changes to the boilerplate code should be performed on a branch of https://github.com/Ouranosinc/cookiecutter-pypackage and merged via a pull request, then applied to this repo using `cruft update`. This will preserve the history, and ensure that this (and any other projects based on this scaffolding) will benefit from changes.

Reviewers:
* Please note any bugs or weirdness by commenting on them.
    --> Changes needed will be performed [here](https://github.com/Ouranosinc/cookiecutter-pypackage).
* Please let me know if the initial values used for cruft are what is desired.
* Do not submit code changes here unless it is adding RavenPy-specific details.

Cheers,

### CODE REVIEW

File: .cruft.json [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+            "pypi_username": "CSHS-CWRA",
+            "version": "0.1.0",
+            "use_pytest": "y",
+            "use_pypi_deployment_with_travis": "y",
+            "add_pyup_badge": "y",
+            "make_docs": "y",
+            "command_line_interface": "Click",
+            "create_author_file": "y",
+            "open_source_license": "MIT license",
+            "_template": "/home/tjs/git/cookiecutter-pypackage"
```
Comment (huard) [inline .cruft.json:21]: Is `_template` with your local home legit here ?

File: CONTRIBUTING.rst [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+Pull Request Guidelines
+-----------------------
+
+Before you submit a pull request, check that it meets these guidelines:
+
+1. The pull request should include tests.
+2. If the pull request adds functionality, the docs should be updated. Put
+   your new functionality into a function with a docstring, and add the
+   feature to the list in README.rst.
+3. The pull request should work for Python 3.5, 3.6, 3.7 and 3.8, and for PyPy. Check
```
Comment (huard) [inline CONTRIBUTING.rst:105]: Remove 3.5 ?

**Review by huard — APPROVED** [review]

File: .cruft.json (hunk 1 above)
Comment (Zeitsperre) [reply .cruft.json:21]: hmmm... Yeah, that's not great. Is it possible to use a URL?

File: CONTRIBUTING.rst (hunk 2 above)
Comment (Zeitsperre) [reply CONTRIBUTING.rst:105]: Yep, that's gotta go. Making a note in a PR now.

Comment (Zeitsperre) [reply CONTRIBUTING.rst:105]: We probably don't want to support PyPy either, I imagine.

File: .cruft.json (hunk 1 above)
Comment (Zeitsperre) [reply .cruft.json:21]: It is! So cool!

**Review by julemai — APPROVED** [review]
I only understand a really small fraction of this. Thanks for all your hard work! --Julie

**Comment by Zeitsperre** [comment]
@huard I think we also want the `cookiecutter.project_slug` to be "RavenPy", right?

**Review by CSHS-Admin — APPROVED** [review]
Looks good to me. --Julie

**Comment by huard** [comment]
@Zeitsperre I think the slug is meant to be lowercase.

File: CONTRIBUTING.rst (hunk 2 above)
Comment (huard) [reply CONTRIBUTING.rst:105]: Correct.

**Comment by Zeitsperre** [comment]
Oh no! Ok.

**Comment by Zeitsperre** [comment]
I was thrown for a loop since the project here is called "RavenPy". Once the library is on PyPI and has documentation, you'll need to update the case spelling in URLs as needed. Outside that, I think this PR is good to go.

File: README.rst [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+Features
+--------
+
+* TODO
+
+Credits
+-------
+
+This package was created with Cookiecutter_ and the `audreyr/cookiecutter-pypackage`_ project template.
```
Comment (tlvu) [inline README.rst:37]: We probably should reference our cookiecutter instead here.

File: setup.cfg [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+commit = True
+tag = True
+
+[bumpversion:file:setup.py]
+search = version='{current_version}'
+replace = version='{new_version}'
+
+[bumpversion:file:ravenpy/__init__.py]
+search = __version__ = '{current_version}'
+replace = __version__ = '{new_version}'
```
Comment (tlvu) [inline setup.cfg:12]: need to add `.cruft.json` to list of files to bump version as well, like here https://github.com/Ouranosinc/raven/blob/58e8978375565fa095e0ec8ae1bf81dc25741f94/setup.cfg#L31-L33

**Review by tlvu — COMMENTED** [review]
Minor changes left, good work with cookiecutter and cruft.

File: README.rst (hunk 3 above)
Comment (Zeitsperre) [reply README.rst]: Done!

**Pull request merged**